lib/nats.js

```diff
@@ -824,10 +843,13 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
   } else {
     psub = 'PUB ' + subject + SPC + opt_reply + SPC;
   }
+
   this.sendCommand(psub + Buffer.byteLength(msg) + CR_LF + msg + CR_LF);
 
   if (opt_callback !== undefined) {
     this.flush(opt_callback);
+  } else if (this.closed) {
+    this.emit('error', PUB_CONN_CLOSED_ERR);
```

Comment:
I would say this emission should be handled by a primary connection lost emission; connection losses themselves aren't request-oriented.  Failures of requests _due_ to connection loss are.  Thus, the callback, and only the callback is an appropriate vehicle for error transmission here.